File: client/php/test/CacheTest.php
<?php

use \PyCached\PyCachedClient;

class CacheTest extends PHPUnit_Framework_TestCase
{
    protected function setUp() {
        $this->client = new PyCachedClient;
        $this->client->connect('localhost', 12345);
        $this->client->clear();
    }

    protected function tearDown() {
        $this->client->close();
    }

    protected function getRandomHash($length) {
        return substr(md5(rand().time()), 0, $length);
    }

    protected function assertCacheCount($expected) {
        $count = $this->client->count();
        $this->assertSame($expected, $count);
    }

    public function testVersion() {
        $response = $this->client->version();
        $this->assertInternalType('string', $response);
    }

    public function testStatus() {
        $status = $this->client->status();
        $this->assertArrayHasKey('uptime', $status);
        $this->assertInternalType('float', $status['uptime']);
    }

    public function testEmpty() {
        $this->assertCacheCount(0);
    }

    public function testClear() {
        $elements = 10;
        foreach(range(0, $elements - 1) as $i)
            $this->client->set($this->getRandomHash(32), $i);
        $this->assertCacheCount($elements);
    }

    public function testSimpleSequence() {
        $this->assertCacheCount(0);

        $this->client->set('john', 'doe');
        $this->assertCacheCount(1);

        $response = $this->client->get('john');
        $this->assertSame('doe', $response);

        $this->client->set('john', 'lennon');
        $this->assertCacheCount(1);

        $response = $this->client->get('john');
        $this->assertSame('lennon', $response);

        $this->client->delete('john');
        $this->assertCacheCount(0);
    }

    public function testNestedStructures() {
        $value = range(1, 10);
        $this->client->set('nested', $value);

        $response = $this->client->get('nested');
        $this->assertSame($response, $value);
    }
}

File: client/python/test/cache_test.py
import unittest
from time import time
import string, random
from client import PyCachedClient

class MainTest (unittest.TestCase):
    def setUp(self):
        self.client = PyCachedClient()
        self.client.connect('localhost', 12345)
        self.client.clear()

    def tearDown(self):
        self.client.close()

    def random_hash(self, length):
        chars = string.ascii_uppercase + string.digits
        return ''.join(random.choice(chars) for x in range(length))

    def assertCacheCount(self, expected):
        count = self.client.count()
        self.assertEqual(expected, count)

    def test_version(self):
        version = self.client.version()
        self.assertTrue(type(version) is unicode)

    def test_empty(self):
        self.assertCacheCount(0)

    def test_clear(self):
        elements = 10
        for i in range(elements):
            self.client.set(self.random_hash(32), i)
        self.assertCacheCount(elements)

    def test_simple_sequence(self):
        self.assertCacheCount(0)

        self.client.set('john', 'doe')
        self.assertCacheCount(1)

        response = self.client.get('john')
        self.assertEqual('doe', response)

        self.client.set('john', 'lennon')
        self.assertCacheCount(1)

        response = self.client.get('john')
        self.assertEqual('lennon', response)

        self.client.delete('john')
        self.assertCacheCount(0)

    def test_nested(self):
        value = range(1, 10)
        self.client.set('nested', value)

        response = self.client.get('nested')
        self.assertEqual(value, response)
